packages/purpose-process/src/services/validators.ts

```diff
@@ -405,3 +407,67 @@ export const verifyRequesterIsConsumerOrDelegateConsumer = async (
     return consumerDelegation?.id;
   }
 };
+
+export const getOrganizationRole = async ({
+  purpose,
+  producerId,
+  delegationId,
+  readModelService,
+  authData,
+}: {
+  purpose: Purpose;
+  producerId: TenantId;
+  delegationId: DelegationId | undefined;
+  readModelService: ReadModelService;
+  authData: UIAuthData | M2MAdminAuthData;
+}): Promise<Ownership> => {
+  if (
+    producerId === purpose.consumerId &&
+    authData.organizationId === producerId
+  ) {
+    return ownership.SELF_CONSUMER;
+  }
+
+  const [producerDelegation, consumerDelegation] = await Promise.all([
+    readModelService.getActiveProducerDelegationByEserviceId(
+      purpose.eserviceId
+    ),
+    retrievePurposeDelegation(purpose, readModelService),
+  ]);
+
+  if (delegationId) {
+    if (delegationId === consumerDelegation?.id) {
+      assertRequesterIsDelegateConsumer(purpose, authData, consumerDelegation);
+      return ownership.CONSUMER;
+    } else if (delegationId === producerDelegation?.id) {
+      assertRequesterIsDelegateProducer(
+        { id: purpose.eserviceId, producerId },
+        authData,
+        producerDelegation
+      );
+      return ownership.PRODUCER;
+    } else {
+      throw tenantNotAllowed(authData.organizationId);
```

Comment:
Also here I would be more specific, the tenant could be allowed but the issue may be in the delegation ID they pass
```suggestion
      throw unauthorizedError("Tenant cannot perform operation as delegate for the specified delegation ID");
```